.github/workflows/playwright.yml

```diff
@@ -128,45 +131,69 @@ jobs:
           PLUGIN_E2E_MOCK_OAUTH_SERVER_URL: http://172.17.0.1:8080
           E2E_TESTING: true
 
-      - name: ci/setup-node
+      # - name: ci/checkout-mattermost-monorepo
+      #   uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3.3.0
+      #   with:
+      #     path: ../mattermost
+      #     repository: mattermost/mattermost
+
+      - name: ci/checkout-mattermost-monorepo
+        run: |
+          git clone https://github.com/mattermost/mattermost.git ../mattermost
```

Comment:
I wonder if the version should be opt-in here. Otherwise changes on `mattermost` could break the E2E tests in this repo. On the other hand, it might be good if we notice a breaking change early.

Reply:
@hanzei This is intentional to have it always use master, though I think we could have two workflows if we also want a pinned version, like we did with the Apps plugin.

> could break the E2E tests in this repo

The main thing that was breaking things in the plugin e2e tests related to the monorepo where the Playwright helpers that the e2e tests were using, like `getLastPost`. The function implementations didn't change (because the general DOM structure of the webapp didn't change), but where the function is located can change at any time. These are also not necessarily meant to be consumed by other projects.

If the plugin e2e test fails because a _behavior_ change in the webapp's functionality, I think we would want to see the test failure to see if it's something we need to address. If this does cause an issue, we'll handle it on a case-by-case basis.

To solve the issue of the helpers changing paths etc., the plan is to have these helpers intended to be used by plugin e2e tests defined in a separate repo, currently housed at https://github.com/mattermosttest/mattermost-plugin-e2e-test-utils/pull/1

Reply:
> This is intentional to have it always use master, though I think we could have two workflows if we also want a pinned version, like we did with the Apps plugin.

:+1: for using the same pattern as the Apps plugin (Release on PR, master on master)